packages/core/src/utils.ts

```diff
@@ -1,3 +1,23 @@
+/**
+ * Create a deep copy (with all fields recursively copied) of an object
+ * @param obj the object to copy
+ * @returns a deep copy of the object
+ */
 export function deepCopy<T>(obj: T): T {
   return JSON.parse(JSON.stringify(obj));
 }
```

Comment:
You haven't touched this, but it's better to use the new-ish `structuredClone()` 

https://developer.mozilla.org/en-US/docs/Web/API/structuredClone

Reply:
Addressed in 9928adf33c7ec8cc57445051fac0db037439fb9d